fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/LivingEntityFeatureRendererRegistrationCallback.java

```diff
@@ -0,0 +1,80 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendering.v1;
+
+import net.minecraft.client.render.entity.EntityRendererFactory;
+import net.minecraft.client.render.entity.LivingEntityRenderer;
+import net.minecraft.client.render.entity.feature.Deadmau5FeatureRenderer;
+import net.minecraft.client.render.entity.feature.FeatureRenderer;
+import net.minecraft.client.render.entity.model.EntityModel;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Called when {@link FeatureRenderer feature renderers} for a {@link LivingEntityRenderer living entity renderer} are registered.
+ *
+ * <p>Feature renderers are typically used for rendering additional objects on an entity, such as armor, an elytra or {@link Deadmau5FeatureRenderer Deadmau5's ears}.
+ * This callback lets developers add additional feature renderers for use in entity rendering.
+ * Listeners should filter out the specific entity renderer they want to hook into, usually through {@code instanceof} checks or filtering by entity type.
+ * Once listeners find a suitable entity renderer, they should register their feature renderer via the registration helper.
+ *
+ * <p>For example, to register a feature renderer for a player model, the example below may used:
+ * <blockquote><pre>
+ * LivingEntityFeatureRendererRegistrationCallback.EVENT.register((entityType, entityRenderer, registrationHelper) -> {
+ * 	if (entityRenderer instanceof PlayerEntityModel) {
+ * 		registrationHelper.register(new MyFeatureRenderer((PlayerEntityModel) entityRenderer));
+ * 	}
+ * });
+ * </pre></blockquote>
+ */
+@FunctionalInterface
+@Environment(EnvType.CLIENT)
+public interface LivingEntityFeatureRendererRegistrationCallback {
+	Event<LivingEntityFeatureRendererRegistrationCallback> EVENT = EventFactory.createArrayBacked(LivingEntityFeatureRendererRegistrationCallback.class, callbacks -> (entityType, entityRenderer, registrationHelper, context) -> {
+		for (LivingEntityFeatureRendererRegistrationCallback callback : callbacks) {
+			callback.registerRenderers(entityType, entityRenderer, registrationHelper, context);
+		}
+	});
+
+	/**
+	 * Called when feature renderers may be registered.
+	 *
+	 * @param entityType     the entity type of the renderer
+	 * @param entityRenderer the entity renderer
+	 */
+	void registerRenderers(EntityType<? extends LivingEntity> entityType, LivingEntityRenderer<?, ?> entityRenderer, RegistrationHelper registrationHelper, EntityRendererFactory.Context context);
+
+	/**
+	 * A delegate object used to help register feature renderers for an entity renderer.
+	 *
+	 * <p>This is not meant for implementation by users of the API.
+	 */
+	interface RegistrationHelper {
```

Comment:
You can add `@ApiStatus.NonExtendable` for this.

Reply:
Done.